src/common/os/posix/mod_loader.cpp

```diff
@@ -69,25 +69,39 @@ bool ModuleLoader::isLoadableModule(const Firebird::PathName& module)
 	return true;
 }
 
-void ModuleLoader::doctorModuleExtension(Firebird::PathName& name)
+bool ModuleLoader::doctorModuleExtension(Firebird::PathName& name, int& step)
```

Comment:
I like an idea of patch but would not it be better to use
class ModuleName : public PathName {.......
containing step number instead adding artificial parameter to doctorModuleExtension function which must be in sync with module's PathName?